ui/appui-react/src/appui-react/layout/footer/message-center/MessageCenterField.tsx

```diff
@@ -0,0 +1,150 @@
+import { MessageCenterMessage } from "./MessageCenterMessage";
+import { MessageManager } from "../../../messages/MessageManager";
+import { TitleBar } from "../dialog/TitleBar";
+import type { NotifyMessageDetailsType } from "../../../messages/ReactNotifyMessageDetails";
+import "./MessageCenterField.scss";
+
+/** Message Center Field React component.
+ * @public
+ */
+export function MessageCenterField() {
+  const [messages, setMessages] = React.useState(MessageManager.messages);
+  const [notify, setNotify] = React.useState("");
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const indicatorRef = React.createRef<HTMLDivElement>();
+  const messageCount = React.useRef(messages.length);
+
+  const title = UiFramework.translate("messageCenter.messages");
+
+  const handleOpenChange = (isOpenState: boolean) => {
+    setNotify("");
+    setIsOpen(isOpenState);
+  };
+
+  const isProblemStatus = (message: NotifyMessageDetailsType): boolean => {
+    // See priority values in DgnPlatform defined in NotificationManager
+    return (
+      message.priority === OutputMessagePriority.Error ||
+      message.priority === OutputMessagePriority.Fatal
+    );
+  };
+
+  const notifyStatus = messages.some((msg) => isProblemStatus(msg))
+    ? "negative"
+    : "primary";
+
+  React.useEffect(() => {
+    MessageManager.registerAnimateOutToElement(indicatorRef.current);
+
+    return MessageManager.onMessagesUpdatedEvent.addListener(() => {
+      setNotify(notifyStatus);
+      setMessages(MessageManager.messages);
+    });
+  });
+
+  const getMessages = (tab: string) => {
+    return [...messages].reverse().map((message, index) => {
+      if (messages.length > 0) {
+        const iconClassName = MessageManager.getIconClassName(message);
+        const iconSpec = MessageManager.getIconSpecFromDetails(message);
+        const text: MessageType = message.briefMessage;
+        if ((tab === "error" && isProblemStatus(message)) || tab === "all") {
+          return (
+            <MessageCenterMessage
+              key={index.toString()}
+              message={text}
+              details={message.detailedMessage}
+              icon={<Icon iconSpec={iconSpec} className={iconClassName} />}
+            />
+          );
+        }
+        // these returns are a work around to avoid 'Not all paths return a value error"
+        return <></>;
+      } else {
+        return (
+          <span className="mc-message-prompt" key={`${index.toString()}`}>
+            No Messages.
+          </span>
+        );
+      }
+    });
+  };
+
+  const tabs = (
+    <>
+      {["all", "error"].map((tabType) => {
+        return (
+          <Tabs.Panel value={tabType} key={tabType}>
+            <div className={"mc-footer-messageCenter-dialog"}>
+              {getMessages(tabType)}
+            </div>
+          </Tabs.Panel>
+        );
+      })}
+    </>
+  );
+
+  return (
+    <>
+      <div ref={indicatorRef} />
```

Comment:
This should work if you move this to line 128 (pass to the `Button` as a `ref` prop. This div would not be needed then. Also if this is moved to the button, the definition on line 35 needs to be changed to `const indicatorRef = React.createRef<HTMLButtonElement>();` The variable name could also be changed but not necessarily.

Reply:
that does work better!